buildscripts/buildvars-setup.sh

```diff
@@ -193,41 +193,44 @@ done
 
 export $__BuildArch
 
+# Use uname to determine what the OS is.
+export OSName=$(uname -s)
+case $OSName in
+    Darwin)
+        export __BuildOS=OSX
```

Comment:
Nit: It may look better to initialize `HostOS` here, and set `BuildOS=HostOS` after the switch statement.